### PR TITLE
Hjem standalone

### LINKED ISSUE #116: Standalone Hjem on Linux distros outside of NixOS
While #4 fixed this issue on darwin by adding `nix-darwin` support, I'd still like to use hjem on a Linux machine that doesn't have NixOS on it.

Not very urgent for me but I'll definitely use it when it's working.

### PR DESCRIPTION
**Pull request by NotAShelf**
As it stands, this PR provides the initial draft of the Hjem CLI I'm envisioning. As [discussed on Matrix](https://matrix.to/#/!VPLNttfHqYLLfpfXnQ:feel-co.org/$Q5inISg5boBJKQwlWkLXagzHIZANhNuBIy5VR_IgzIA?via=feel-co.org&via=matrix.org&via=notashelf.dev). The CLI currently provides three invocation methods (`hjem`, `hjem internal` and `hjem standalone`) and acts as a replacement for `pkgs.smfh`, leveraging its library component after my latest refactor to the [smfh crate](https://crates.io/smfh-core).

Since we're using *Rust*, I've decided that the CLI should replace a fair bit of NixOS and Darwin logic, primarily the Bash parts, in order to provide necessary primitives for standalone and a more polished base for the existing platform support. The interface for standalone which (for the time being) includes `hjem standalone init`, `hjem standalone build`, `hjem standalone switch` and a few other goodies (see `--help`). The CLI is not final, because I have not yet decided whether I like the subcommand approach. There's limited work done on making the `hjem` binary a multi-call binary in order to make its usage more robust. The names and design choices are all WIP, and feedback is welcome. 

Tests, documentation and modules all updated. I'd appreciate testing on real systems as VM tests don't provide enough validation against edge cases.

### CODE REVIEW

File: cli/crates/hjem-core/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,1812 @@
...
+        return Ok(PathBuf::from(config_home).join("hjem"));
+    }
+    Ok(home_dir()?.join(".config").join("hjem"))
+}
+
+fn home_dir() -> Result<PathBuf, String> {
+    std::env::var("HOME")
+        .map(PathBuf::from)
+        .map_err(|_| "HOME is not set".to_string())
+}
```
Comment (HigherOrderLogic) [inline cli/crates/hjem-core/src/lib.rs:1745]: There's [this](https://doc.rust-lang.org/std/env/fn.home_dir.html), which I think would be better here.

Comment (NotAShelf) [reply cli/crates/hjem-core/src/lib.rs:1745]: I keep forgetting this is a thing lol. I'll change, thanks for the reminder.

**Comment by Gerg-L** [comment]
Huuuuuuuge
